DylibBinder/StreamWriterExtensions.cs

```diff
@@ -0,0 +1,887 @@
+				HandleAssociateBoundType (parameters as SwiftReflector.SwiftBoundGenericType, ref associatedTypeList);
+				break;
+			default:
+				break;
+			}
+
+			return associatedTypeList;
+		}
+
+		static void HandleAssociateTupleType (SwiftReflector.SwiftTupleType parameters, ref List<string> associatedTypeList)
+		{
+			foreach (var contents in parameters.Contents) {
+				if (contents is SwiftReflector.SwiftGenericArgReferenceType)
+					HandleAssociateGenericArgType (contents as SwiftReflector.SwiftGenericArgReferenceType, ref associatedTypeList);
+				else if (contents is SwiftReflector.SwiftBoundGenericType)
+					HandleAssociateBoundType (contents as SwiftReflector.SwiftBoundGenericType, ref associatedTypeList);
+				else if (contents is SwiftReflector.SwiftFunctionType)
+					HandleAssociateFunctionType (contents as SwiftReflector.SwiftFunctionType, ref associatedTypeList);
+			}
+		}
+
+		static void HandleAssociateBoundType (SwiftReflector.SwiftBoundGenericType contents, ref List<string> associatedTypeList)
+		{
+			foreach (var boundTypes in contents.BoundTypes) {
+				var associatedGenericArgTypes = boundTypes as SwiftReflector.SwiftGenericArgReferenceType;
+				if (associatedGenericArgTypes == null)
+					continue;
+				HandleAssociateGenericArgType (associatedGenericArgTypes, ref associatedTypeList);
+			}
+		}
+
+		static void HandleAssociateFunctionType (SwiftReflector.SwiftFunctionType parameter, ref List<string> associatedTypeList)
+		{
+			if (parameter.Parameters is SwiftReflector.SwiftGenericArgReferenceType) {
+				HandleAssociateGenericArgType (parameter.Parameters as SwiftReflector.SwiftGenericArgReferenceType, ref associatedTypeList);
+			} else if (parameter.Parameters is SwiftReflector.SwiftBoundGenericType) {
+				HandleAssociateBoundType (parameter.Parameters as SwiftReflector.SwiftBoundGenericType, ref associatedTypeList);
+			}
+		}
+
+		static void HandleAssociateGenericArgType (SwiftReflector.SwiftGenericArgReferenceType parameter, ref List<string> associatedTypeList)
+		{
+			foreach (var associatedType in parameter.AssociatedTypePath) {
+				if (!associatedTypeList.Contains (associatedType)) {
+					associatedTypeList.Add (associatedType);
+				}
+			}
+		}
+
+		static List<string> WriteReturnType (this StreamWriter sw, SwiftReflector.SwiftBaseFunctionType signature, int depth, int genericArgumentsCount = 0, int genericParameterCount = 0)
+		{
+			if (signature == null)
+				return null;
+
+			var returnType = signature.ReturnType;
+			if (returnType != null && !String.IsNullOrEmpty (returnType.ToString ()) && returnType.ToString () != "()") {
+				var enhancedReturn = StringBuilderHelper.EnhanceReturn (signature, depth, genericArgumentsCount, genericParameterCount);
+				var escapedReturn = StringBuilderHelper.EscapeCharacters (enhancedReturn);
+				if (escapedReturn != null) {
+					sw.WriteTypeValue ("returnType", escapedReturn);
+					var retList = new List<string> ();
+					if (escapedReturn.Contains ("T0"))
```

Comment:
What is this for?

Reply:
This is a part of my not very good way of tracking generic parameters inside of typedeclarations. I write the methods and properties of the a typedeclaration and then while I write them, I keep track if I wrote a "T0" or "T1" and then pass that up to the typedeclaration level to write the generic parameters of the typedeclaration. 
To make this a lot simpler, the options are either run over the the typedeclarations using an XDocument and adding them in and/or just creating the whole thing with an XmlWriter and being able to read them as I write them!